storm/src/main/storm/mesos/MesosSupervisor.java

```diff
@@ -107,7 +110,7 @@ public Object getMetadata() {
     for (int i = 0; i < ports.length; i++) {
       p[i] = Integer.parseInt((String) ports[i]);
     }
-    return PersistentVector.create(p);
+    return PersistentVector.create((Object[]) p);
```

Comment:
Also, just for accuracy, this is "upcasting", not "downcasting".  Downcasting means converting to a more specific derived type:
* https://en.wikipedia.org/wiki/Downcasting